src/NuGet.Core/NuGet.Configuration/Settings/SettingFactory.cs

```diff
@@ -99,10 +100,30 @@ internal static SettingBase Parse(XNode node, SettingsFile origin)
         internal static IEnumerable<T> ParseChildren<T>(XElement xElement, SettingsFile origin, bool canBeCleared) where T : SettingElement
         {
             var children = new List<T>();
+            IEnumerable<T> descendants = xElement.Elements().Select(d => Parse(d, origin)).OfType<T>();
+            HashSet<T> distinctDescendants = new HashSet<T>();
 
-            var descendants = xElement.Elements().Select(d => Parse(d, origin)).OfType<T>().Distinct();
+            List<T> duplicatedDescendants = new List<T>();
 
-            foreach (var descendant in descendants)
+            foreach (var item in descendants)
+            {
+                if (!distinctDescendants.Add(item))
```

Comment:
How does the Hashset know what to use as the key?  Does the object coming back from `Parse()` implement comparison?

Reply:
No, the object does not implement comparison, I created a new comparer